test/cypress/integration/simplified-account/service-settings/switch-psp/switch-worldpay-credentials/switch-worldpay-credentials.cy.js

```diff
@@ -0,0 +1,419 @@
+      gateway_account_id: GATEWAY_ACCOUNT_ID,
+      type: LIVE_ACCOUNT_TYPE,
+      payment_provider: WORLDPAY,
+      provider_switch_enabled: options.switchEnabled,
+      allow_moto: options.moto,
+      worldpay_3ds_flex: opts.worldpay_3ds_flex || undefined,
+      gateway_account_credentials: options.accountCredentials
+    }),
+    ...additionalStubs])
+}
+
+describe('Switch Worldpay credentials setting', () => {
+  beforeEach(() => {
+    cy.setEncryptedCookies(USER_EXTERNAL_ID)
+  })
+
+  describe('The settings nav', () => {
+    beforeEach(() => {
+      setupStubs({}, [])
+      cy.visit(SWITCH_TO_WORLDPAY_SETTINGS_URL)
+    })
+    it('should show active "Switch Worldpay credentials" link', () => {
+      checkSettingsNavigation('Switch Worldpay credentials', SWITCH_TO_WORLDPAY_SETTINGS_URL)
+    })
+  })
+
+  describe('The page', () => {
+    beforeEach(() => {
+      setupStubs({}, [])
+      cy.visit(SWITCH_TO_WORLDPAY_SETTINGS_URL)
+    })
+    it('should have the correct title and heading', () => {
+      checkTitleAndHeading('Switch Worldpay credentials', SERVICE_NAME.en)
+    })
+  })
+
+  describe('For a non-admin', () => {
+    beforeEach(() => {
+      setupStubs({
+        role: 'view-and-refund',
+        moto: true
+      }, [])
+      cy.visit(SWITCH_TO_WORLDPAY_SETTINGS_URL, { failOnStatusCode: false })
+    })
+
+    it('should show admin only error', () => {
+      cy.title().should('eq', 'An error occurred - GOV.UK Pay')
+      cy.get('h1').should('contain.text', 'An error occurred')
+      cy.get('#errorMsg').should('contain.text', 'You do not have the administrator rights to perform this operation.')
+    })
+  })
+
+  describe('For an admin', () => {
+    describe('For a non-MOTO service', () => {
+      describe('When no tasks have been completed', () => {
+        beforeEach(() => {
+          setupStubs()
+          cy.visit(SWITCH_TO_WORLDPAY_SETTINGS_URL)
+        })
+        it('The task list should have the tasks required for switching Worldpay credentials', () => {
+          checkDisplayedTasks([
+            {
+              name: 'Link your Worldpay account with GOV.UK Pay',
+              status: taskStatus.NOT_STARTED,
+              tagClass: 'govuk-tag govuk-tag--blue'
+            },
+            {
+              name: 'Configure 3DS',
+              status: taskStatus.NOT_STARTED,
+              tagClass: 'govuk-tag govuk-tag--blue'
+            },
+            {
+              name: 'Make a live payment to test your Worldpay PSP',
+              status: taskStatus.CANNOT_START,
+              tagClass: 'govuk-tag govuk-tag--grey'
+            }
+          ])
+        })
+      })
+
+      describe('When worldpay credentials and 3DS flex tasks have been completed', () => {
+        beforeEach(() => {
+          setupStubs({
+            moto: false,
+            worldpay_3ds_flex: {
+              organisational_unit_id: '5bd9b55e4444761ac0af1c80',
```

Comment:
can we use the consts defined in ln 9:11 here?
or use the `VALID_FLEX_CREDENTIALS` object?